scripts/build_shapes.py

```diff
@@ -640,25 +647,41 @@ def _process_func_pop(c_code):
 
     # get worldpop image
     WorldPop_inputfile, WorldPop_filename = download_WorldPop(
-            c_code, year, False, False)
+        c_code, year, False, False)
 
     with rasterio.open(WorldPop_inputfile) as src:
 
         for i, row in country_rows.iterrows():
-            country_rows.loc[i, "pop"] = _sum_raster_over_mask(row.geometry, src)
-                    
+            country_rows.loc[i,
+                             "pop"] = _sum_raster_over_mask(row.geometry, src)
+
     return country_rows
 
 
-def add_population_data(df_gadm,
-                        country_codes,
-                        year=2020,
-                        update=False,
-                        out_logging=False,
-                        nprocesses=2,
-                        disable_progressbar=False
-    ):
-    """Function to add the population info for each country shape in the gadm dataset"""
+def add_population_data(
+    df_gadm,
+    country_codes,
```

Comment:
To answer your question, country_codes are needed to download the appropriate worldpop data by country, whereas in gdp only a file is considered, thus they are not needed.
We could get rid of the mandadory column, but the same information must be got in other ways. Options:
- by a predefined dataframe column name [e.g. country]
- by a column name specified as an optional parameter of the function [maybe preferred option as more general]

The above can be addressed also in other moments, not necessarily in this pull requested, or never if not considered useful

Reply:
Yeah makes sense. Agree, let's discuss that another time